native/core/src/parquet/mod.rs

```diff
@@ -703,6 +704,7 @@ pub unsafe extern "system" fn Java_org_apache_comet_parquet_Native_initRecordBat
             file_groups,
             None,
             data_filters,
+            pushdown_filters != 0,
```

Comment:
Same here. `data_filters` is an `Option` and should be `None` if filter pushdown is disabled.